src/Environment/AbstractConfigurableEnvironment.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace UnicornFail\Emoji\Environment;
+
+use UnicornFail\Emoji\Configuration\Configuration;
+use UnicornFail\Emoji\Configuration\ConfigurationAwareInterface;
+use UnicornFail\Emoji\Traits\ConfigurableEnvironmentTrait;
+
+class AbstractConfigurableEnvironment extends AbstractEnvironment implements ConfigurableEnvironmentInterface
```

Comment:
All of these "Environment" based abstractions would be intended to go into something like `league/environment` or even part of `league/ast` or have the latter provide additional "renderable" aspects.

The goal was to split out the environment components into separate interfaces and traits, so it could be "built-up" as needed. Still not 💯 there, but I hope this gives an idea of what I'm after and what could be easier to implement in the long run for both projects.

Reply:
This has been simplified to use `league/configuration` (https://github.com/unicorn-fail/configuration)